Index: Rendering/vtkRenderer.cxx
===================================================================
RCS file: /cvsroot/VTK/VTK/Rendering/vtkRenderer.cxx,v
retrieving revision 1.240
diff -u -r1.240 vtkRenderer.cxx
--- Rendering/vtkRenderer.cxx	16 Apr 2008 18:46:05 -0000	1.240
+++ Rendering/vtkRenderer.cxx	13 Jun 2008 12:45:18 -0000
@@ -1758,6 +1758,21 @@
       //should never get here
       return 0; 
     }        
+  // Check if there is translucent geometry
+  // do the render library specific stuff about translucent polygonal geometry.
+  // As it can be expensive, do a quick check if we can skip this step
+  int hasTranslucentPolygonalGeometry=0;
+  for ( i = 0; !hasTranslucentPolygonalGeometry && i < this->PropArrayCount;
+        i++ )
+    { 
+    hasTranslucentPolygonalGeometry=
+      this->PropArray[i]->HasTranslucentPolygonalGeometry();
+    }
+  // Have to be set before a call to UpdateTranslucentPolygonalGeometry()
+  // because UpdateTranslucentPolygonalGeometry() will eventually call
+  // vtkOpenGLActor::Render() that uses this flag.
+  if(hasTranslucentPolygonalGeometry)
+    this->LastRenderingUsedDepthPeeling=0;
 
   // loop through props and give them a chance to 
   // render themselves as opaque geometry
@@ -1786,6 +1801,8 @@
       {
       this->NumberOfPropsRendered += 
         this->PropArray[i]->RenderOpaqueGeometry(this);
+      if(hasTranslucentPolygonalGeometry != 0 && this->PropArray[i]->HasTranslucentPolygonalGeometry())
+        this->PropArray[i]->RenderTranslucentPolygonalGeometry(this);
       }
 
     //restore the prop's original settings
