--- /tmp/app.sh	2010-11-30 17:27:08.692139067 +0100
+++ jcae	2010-11-30 17:36:36.976138104 +0100
@@ -72,6 +72,16 @@
     clusters="$clusters:$extra_clusters"
 fi
 
+#An absolut path is prefered as sub-processes may be run from other directories
+curdir="$PWD"
+cd "$progdir/../$APPNAME/modules/lib"
+libdir="$PWD"
+cd "$curdir"
+LD_LIBRARY_PATH="$libdir:$LD_LIBRARY_PATH"
+export LD_LIBRARY_PATH
+MMGT_OPT=0
+export MMGT_OPT
+
 nbexec=`echo "$progdir"/../platform*/lib/nbexec`
 
 case "`uname`" in
